tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/V3/XUnitTestMethodRunnerBaseRunTestCaseV3Integration.cs

```diff
@@ -0,0 +1,243 @@
+
+                    var mrunner = instance.DuckCast<IXunitTestMethodRunnerV3>()!;
+
+                    // Decrement the execution number (the method body will do the execution)
+                    retryMetadata.ExecutionNumber--;
+                    var innerReturnValue = (TReturn)await mrunner.RunTestCase(context.Instance!, testcase.Instance!);
+                    Common.Log.Debug<int, int, string>("EFD/Retry: [Retry {Num}] Retry finished. [Current retry value is {Value}]. DisplayName: {DisplayName}", retryNumber, retryMetadata.ExecutionNumber, testcase.TestCaseDisplayName);
+
+                    var innerReturnValueUnsafe = Unsafe.As<TReturn, RunSummaryUnsafeStruct>(ref innerReturnValue);
+                    Common.Log.Debug<int>("EFD/Retry: [Retry {Num}] Aggregating results.", retryNumber);
+                    runSummaryUnsafe.Total += innerReturnValueUnsafe.Total;
+                    runSummaryUnsafe.Failed += innerReturnValueUnsafe.Failed;
+                    runSummaryUnsafe.Skipped += innerReturnValueUnsafe.Skipped;
+                    runSummaryUnsafe.NotRun += innerReturnValueUnsafe.NotRun;
+                    runSummaryUnsafe.Time += innerReturnValueUnsafe.Time;
+                }
+            }
+            else
+            {
+                if (isFlakyRetryEnabled && runSummaryUnsafe.Failed == 0)
+                {
+                    Common.Log.Debug("EFD/Retry: [FlakyRetryEnabled] A non failed test execution was detected.");
+                }
+                else
+                {
+                    Common.Log.Debug("EFD/Retry: All retries were executed.");
+                }
+            }
+
+            if (index == 0)
+            {
+                retryMessageBus.FlushMessages(retryMetadata.UniqueID);
+
+                // Let's clear the failed and skipped runs if we have at least one successful run
+#pragma warning disable DDLOG004
+                Common.Log.Debug($"EFD/Retry: Summary: {testcase.TestCaseDisplayName} [Total: {runSummaryUnsafe.Total}, Failed: {runSummaryUnsafe.Failed}, Skipped: {runSummaryUnsafe.Skipped}]");
+#pragma warning restore DDLOG004
+                var passed = runSummaryUnsafe.Total - runSummaryUnsafe.Skipped - runSummaryUnsafe.Failed;
+                if (passed > 0)
+                {
+                    runSummaryUnsafe.Total = 1;
+                    runSummaryUnsafe.Failed = 0;
+                    runSummaryUnsafe.Skipped = 0;
+                }
+                else if (runSummaryUnsafe.Skipped > 0)
+                {
+                    runSummaryUnsafe.Total = 1;
+                    runSummaryUnsafe.Skipped = 1;
+                    runSummaryUnsafe.Failed = 0;
+                }
+                else if (runSummaryUnsafe.Failed > 0)
+                {
+                    runSummaryUnsafe.Total = 1;
+                    runSummaryUnsafe.Skipped = 0;
+                    runSummaryUnsafe.Failed = 1;
+                }
+
+#pragma warning disable DDLOG004
```

Comment:
Just flaunting the rules, so blatant 😂 😛

Reply:
removed the warning here https://github.com/DataDog/dd-trace-dotnet/pull/6573/commits/0bb1ed8bf288df45f34491c03669193dd5f59b8b